lib/Entities/NewsletterEntity.php

```diff
@@ -257,6 +259,35 @@ public function getStatus() {
    */
   public function setStatus($status) {
     $this->status = $status;
+
+    // activate/deactivate unfinished tasks
+    $newTaskStatus = null;
+    $typesWithActivation = [self::TYPE_NOTIFICATION, self::TYPE_WELCOME, self::TYPE_AUTOMATIC];
+
+    if (($status === self::STATUS_DRAFT) && in_array($this->type, $typesWithActivation)) {
+      $newTaskStatus = ScheduledTaskEntity::STATUS_PAUSED;
+    }
+    if (($status === self::STATUS_ACTIVE) && in_array($this->type, $typesWithActivation)) {
+      $newTaskStatus = ScheduledTaskEntity::STATUS_SCHEDULED;
+    }
+
+    if (!$newTaskStatus) return;
+
+    $queues = $this->getUnfinishedQueues();
+
+    foreach ($queues as $queue) {
+      /** @var SendingQueueEntity $queue */
+      $task = $queue->getTask();
+      if ($task === null) continue;
+
+      $scheduled = new Carbon($task->getScheduledAt());
+      if ($scheduled < (new Carbon())->subDays(30)) return;
+
+      if (($status === self::STATUS_DRAFT) && ($task->getStatus() !== ScheduledTaskEntity::STATUS_SCHEDULED)) return;
+      if (($status === self::STATUS_ACTIVE) && ($task->getStatus() !== ScheduledTaskEntity::STATUS_PAUSED)) return;
+
```

Comment:
I've just noticed that in these 3 conditions we use `return` but there should be `continue` or we may skip some tasks.